src/mutation_data.cpp

```diff
@@ -725,11 +725,20 @@ void mutation_branch::check_consistency()
         }
         for( const trait_id &addition : mdata.additions ) {
             const mutation_branch &adata = addition.obj();
+
+            // Skip obsolete additions aka additions without any category.
+            if( adata.category.empty() ) {
+                continue;
```

Comment:
```suggestion
            if( mdata.category.empty() ) {
                break;
```

This is equivalent to previous behavior in regard to those obsoleted mutations and also generally a little bit safer (will catch a mistake in which a mutation somehow ends up without category but still has `leads_to` from a normal mutation)